main.go

```diff
@@ -85,13 +90,15 @@ func main() {
 		if ext := path.Ext(r.URL.Path); ext == "" && r.URL.Path != "/" {
 			r.URL.Path += ".html"
 		}
-		fs.ServeHTTP(w, r)
+		metricsManager.Middleware(fs.ServeHTTP)(w, r)
 	})
-	http.HandleFunc("/api/kratos/self-service/login/browser", handleCreateFlow)
-	http.HandleFunc("/api/kratos/self-service/login/flows", handleLoginFlow)
-	http.HandleFunc("/api/kratos/self-service/login", handleUpdateFlow)
-	http.HandleFunc("/api/kratos/self-service/errors", handleKratosError)
-	http.HandleFunc("/api/consent", handleConsent)
+	//status_code.ResponseWithStatusCodeMiddleware()
+	http.HandleFunc("/api/kratos/self-service/login/browser", metricsManager.Middleware(handleCreateFlow))
+	http.HandleFunc("/api/kratos/self-service/login/flows", metricsManager.Middleware(handleLoginFlow))
+	http.HandleFunc("/api/kratos/self-service/login", metricsManager.Middleware(handleUpdateFlow))
+	http.HandleFunc("/api/kratos/self-service/errors", metricsManager.Middleware(handleKratosError))
+	http.HandleFunc("/api/consent", metricsManager.Middleware(handleConsent))
+	http.HandleFunc(prometheus.PrometheusPath, metricsManager.Middleware(prometheus.PrometheusMetrics))
```

Comment:
so the fact that we are using the DefualtServeMux means that we are missing out on loads of features coming from "router" libs, like the `Use` method on chi, gorilla/mux etc

Might be useful having a chat before merging this